Kernel/SharedInterruptHandler.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <AK/Types.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/HashTable.h>
+
+class InterruptHandler;
+class SharedInterruptHandler final {
+public:
+
+    static void initialize(u8 interrupt_number);
+    static SharedInterruptHandler& get_shared_interrupt_handler(u8 interrupt_number);
+    ~SharedInterruptHandler();
+    void handle_interrupt();
+
+    u8 interrupt_number() const { return m_interrupt_number; }
+
+    void register_interrupt_handler(InterruptHandler&);
+    void unregister_interrupt_handler(InterruptHandler&);
+
+protected:
```

Comment:
`protected` is not meaningful in a `final` class since it cannot be subclassed. These can just be `private`.